Umee-Bots/lendingPool-reentrancy/src/agent.ts

```diff
@@ -0,0 +1,38 @@
+import { Finding, HandleTransaction, TransactionEvent } from "forta-agent";
+
+import utils from "./utils";
+
+export const handleTransaction =
+  (contractAddress: string, reenterancyFunctionsSignatures: string[]): HandleTransaction =>
+  async (txEvent: TransactionEvent) => {
+    const findings: Finding[] = [];
+    const { traces } = txEvent;
+    for (let i = 0; i < traces.length; i++) {
+      const trace = traces[i];
+      const depth = trace.traceAddress.length;
+
+      if (trace.action.to === contractAddress) {
+        let j;
+
+        for (j = i + 1; j < traces.length; j++) {
+          if (traces[j].traceAddress.length === depth) {
+            i = j;
+            break; // subtree ended
+          }
+          if (traces[j].action.to === contractAddress) {
+            reenterancyFunctionsSignatures.map((functionSignature) => {
+              try {
+                utils.FUNCTIONS_INTERFACE.decodeFunctionData(functionSignature.split("(")[0], txEvent.transaction.data);
+                findings.push(utils.createFinding(functionSignature.split("(")[0]));
+              } catch (e) {}
+            });
```

Comment:
This almost works, but you're examining the transaction (not the trace) data to get the signature.
Here's my suggestion:
- Create `REENTRANCY_FUNCTIONS_SELECTORS`, like this (also fix the typo "reenterancy" while you're at it):
```TS
// utils.ts
const REENTRANCY_FUNCTIONS_SELECTORS = Object.values(FUNCTIONS_INTERFACE.functions).map(el => iface.getSighash(el));
```
- In this part I selected, instead do:
```TS
const selector = (traces[j].action.input || "").slice(0, 10); // "0x" and first 4 bytes
if (REENTRANCY_FUNCTIONS_SELECTORS.includes(selector)) {
  // push finding
  break; // no need to check other traces, we just need to know if one case happened
}
```